src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1334,6 +1336,95 @@ select Task.Factory.StartNew(() => pair.Item1.Receive(new byte[1]), Cancellation
         }
     }
 
+    public class BlockingSendReceive
+    {
+
+        private readonly ITestOutputHelper _log;
+        public static readonly object[][] Loopbacks = MemberDatas.Loopbacks;
+
+        public BlockingSendReceive(ITestOutputHelper output)
+        {
+            _log = output;
+        }
+
+        [Theory]
+        [MemberData(nameof(Loopbacks))]
+        public async Task BlockingStreamRead_Close(IPAddress ipAddress)
+        {
+            // Test for #22564 when close is called while blocking read is pending.
+            using (var server = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            using (var client = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            {
+                server.BindToAnonymousPort(ipAddress);
+                server.Listen(1);
+                byte[] buffer = new byte[1];
+                buffer[0] = Convert.ToByte('a');
```

Comment:
I don't think you need to use Convert here, you can just cast to byte, i.e. `(byte)'a'` -- we do this in product code in some places.